src/QRCoders.jl

```diff
@@ -242,4 +316,119 @@ function exportqrcode( message::AbstractString
                              width=width)
     exportbitmat(matrix, path; targetsize=targetsize)
 end
+
+# new API for qrcode and exportqrcode
+"""
+    qrcode(code::QRCode)
+
+Create a QR code matrix by the `QRCode` object.
+
+Note: It would raise an error if failed to use the specified `mode`` or `version`.
+"""
+function qrcode(code::QRCode)
+    # raise error if failed to use the specified mode or version
+    mode, eclevel, version, mask = code.mode, code.eclevel, code.version, code.mask
+    message, width = code.message, code.width
+    getmode(message) ⊆ mode || throw(EncodeError("Mode $mode can not encode the message"))
+    getversion(message, mode, eclevel) ≤ version ≤ 40 || throw(EncodeError("The version $version is too small"))
+
+    # encode message
+    data = encodemessage(message, mode, eclevel, version)
+
+    # Generate qr code matrix
+    matrix = emptymatrix(version)
+    maskmat = makemask(matrix, mask)
+    matrix = placedata!(matrix, data) # fill in data bits
+    addversion!(matrix, version) # fill in version bits
+    matrix = addformat!(xor.(matrix, maskmat), mask, eclevel)
+
+    # white border
+    width == 0 && return matrix
+    background = falses(size(matrix) .+ (width*2, width*2))
+    background[width+1:end-width, width+1:end-width] = matrix
+    return background
+end
+
+"""
+    exportqrcode( code::QRCode
+                , path::AbstractString = "qrcode.png"
+                ; targetsize::Int = 5)
+
+Create an image with the encoded `message` of approximate size `targetsize`.
+"""
+function exportqrcode( code::QRCode
+                     , path::AbstractString = "qrcode.png"
+                     ; targetsize::Int = 5)
+    matrix = qrcode(code)
+    exportbitmat(matrix, path; targetsize = targetsize)
+end
+
+"""
+    exportqrcode( codes::AbstractVector{QRCode}
+                , path::AbstractString = "qrcode.gif"
+                ; targetsize::Int = 5
+                , fps::Int = 2)
+
+Create an animated gif with `codes` of approximate size `targetsize`.
+
+The frame rate `fps` is the number of frames per second.
+
+Note: The `codes` should have the same size while the other properties can be different.
+"""
+function exportqrcode( codes::AbstractVector{QRCode}
+                     , path::AbstractString = "qrcode.gif"
+                     ; targetsize::Int = 5
+                     , fps::Int = 2)
+    # check whether the image format is supported
+    if !endswith(path, r"\.\w+")
+        path *= ".gif"
+    else
+        ext = last(split(path, '.'))
+        ext == "gif" || throw(EncodeError(
+            "$ext\n is not a valid format for animated images"))
+    end
+    # code = cat(qrcode.(codes)...; dims=3)
+    n, _ = shape(first(codes))
+    scale = ceil(Int, 72 * targetsize / 2.45 / n)
+    code = Array{Bool}(undef, n * scale, n * scale, length(codes))
+    for (i, c) in enumerate(codes)
+        mat = qrcode(c)
+        code[:,:,i] = kron(mat, trues(scale, scale))
```

Comment:
maybe make a new helper function here because `kron` isn't always clear to readers what it does:

```julia
function _imresize(mat, sz)
    scale = ceil(Int, sz / 2.45 / n)
    kron(mat, trues(scale, scale))
end
```